11 - Stellar Burgers: реализация проекта

DaryaSave opened the pull request.

Commit:
11 - Stellar Burgers

Commit:
11

Commit:
11 - Stellar Burgers проект

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)


Работа проделана огромная:




* Отлично, что в `prepare` создаете `id`
* Отлично сделана логика отображения попапа и страницы  под ним
* Хорошо сделали `ProtectedRoute`


но есть некоторые недочеты:


* Проект не запускается. Скрин https://disk.yandex.ru/i/K4nDTcTDCK9jMQ



Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

Commit:
Исправление всех ошибок и предупреждений в консоли

🔧 React компоненты:
- Создан InputWrapper для устранения предупреждений onPointerCapture
- Исправлены все формы ввода (login, register, profile, forgot/reset password)

⚙️ Webpack конфигурация:
- Добавлен publicPath: '/' для корректной работы SPA маршрутизации
- Исправлены 404 ошибки при загрузке bundle.js на подстраницах

🔐 Аутентификация:
- Реализован checkUserAuth thunk для проверки токенов перед API запросами
- Улучшена обработка 403 ошибок и автоматическое обновление токенов

🌐 WebSocket соединения:
- Добавлен urlGenerator для динамического обновления токенов в WSS
- Улучшена обработка ошибок подключения и переподключения
- Увеличено количество попыток переподключения

🧭 React Router:
- Добавлены future флаги v7_startTransition и v7_relativeSplatPath
- Устранены предупреждения о миграции на v7

📱 Пользовательский интерфейс:
- Оптимизированы useEffect в feed.tsx и profile-orders.tsx
- Улучшена стабильность загрузки данных

Commit:
справлены проблемы со сборкой проекта

🔧 обавлены недостающие зависимости:
- core-js@^3.37.1 - устранение ошибок импорта es.modules
- @storybook/react@^7.6.5 - поддержка TypeScript типов для Storybook

📦 роблемы которые решены:
- Module not found: core-js/modules/es.array.concat.js
- Cannot find module '@storybook/react' TS2307
- роект теперь собирается с чистого npm install

✅ ротестировано: npm install && npm run build проходит успешно

Commit:
fix(ws): correct WS base URL to /orders and cleanup drafts; prod build ok

Commit:
chore(tsconfig): remove excludes for deleted drafts

Commit:
build: add core-js dep; exclude storybook stories from TS and webpack build

Commit:
feat(header): highlight Feed link on /feed and nested routes

Commit:
feat(header): add logout dispatch to mobile menu

Commit:
fix(auth): автообновление accessToken и нормализация Bearer; retry на 401/403\nstyle(header): убрано подчёркивание, «ыход» как «рофиль»\nchore: типизация WS, удаление debug

Commit:
chore: remove all code and css comments across src/public

Commit:
chore(build): TS 5.5.4 for eslint compatibility; code-splitting with hashed filenames; analyzer opt-in; fix warnings

Commit:
финальные правки
